cassandra/deserializers.pyx

```diff
@@ -127,8 +127,8 @@ cdef class DesCounterColumnType(DesLongType):
 
 cdef class DesDateType(Deserializer):
     cdef deserialize(self, Buffer *buf, int protocol_version):
-        cdef double timestamp = unpack_num[int64_t](buf) / 1000.0
```

Comment:
Your concern make sense Adam.  The initial issue with this is basically a double rounding issue. 

So for a date specified by the user: datetime.datetime(2038, 12, 31, 10, 10, 10, 15021). Our Cython extension returns after deserialization.

Let me dig a little bit more and understand why the python impl doesn't have this issue.